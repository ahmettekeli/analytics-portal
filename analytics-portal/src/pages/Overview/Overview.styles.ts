import styled from "styled-components";
import { motion } from "framer-motion";

export const StyledMotionDiv = styled(motion.div)`
  width: 90%;
`;
