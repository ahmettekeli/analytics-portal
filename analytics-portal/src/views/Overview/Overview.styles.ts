import styled from "styled-components";
//TODO Overview page styling.
