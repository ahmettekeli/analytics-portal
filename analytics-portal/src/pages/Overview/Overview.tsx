import { StyledMotionDiv } from "./Overview.styles";
import CardContainer from "components/CardContainer/CardContainer";

function Overview() {
  return (
    <StyledMotionDiv
      initial={{ opacity: 0 }}
      animate={{ opacity: 1 }}
      exit={{ opacity: 0 }}
    >
      <CardContainer data-testid="Overview" />
    </StyledMotionDiv>
  );
}

export default Overview;
