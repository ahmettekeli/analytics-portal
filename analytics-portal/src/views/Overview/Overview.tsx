import { StyledMotionDiv } from "./Overview.styles";
import CardContainer from "../../components/CardContainer/CardContainer";

function Overview() {
  return (
    <StyledMotionDiv>
      <CardContainer />
    </StyledMotionDiv>
  );
}

export default Overview;
