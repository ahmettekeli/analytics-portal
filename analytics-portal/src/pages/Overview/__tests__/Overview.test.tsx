import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Overview from "../Overview";

describe("Overview", () => {
  let renderResult: ReturnType<typeof render>;
  beforeEach(() => {
    renderResult = render(<Overview />);
  });
  test("Renders Overview component", () => {
    expect(renderResult.getByTestId("Overview")).toBeInTheDocument();
  });

  test("Renders CardContainer component", () => {
    expect(renderResult.getByTestId("CardContainer")).toBeInTheDocument();
  });

  test("Overview matches snapshot", () => {
    const tree = renderer.create(<Overview />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
