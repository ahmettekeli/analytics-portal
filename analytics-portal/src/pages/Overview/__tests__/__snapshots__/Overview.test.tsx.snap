// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Overview matches snapshot 1`] = `
<div
  className="sc-bdvvtL eILeqO"
  data-testid="Overview"
  style={Object {}}
>
  <div
    className="sc-dJjYzT kJDsuI"
    data-testid="CardContainer"
  >
    <div
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        style={Object {}}
      />
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        style={Object {}}
      />
    </div>
  </div>
</div>
`;
